### PR TITLE
deposit: add more fields

### PR DESCRIPTION
**Pull request by zzacharo**
Depends on https://github.com/inveniosoftware/react-invenio-deposit/pull/73
Depends on https://github.com/inveniosoftware/react-invenio-forms/pull/28

### CODE REVIEW

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js [hunk 1]
```diff
@@ -101,20 +113,79 @@ export class RDMDepositForm extends Component {
...
+            { text: "ISVARIANTFORMOF", value: "isvariantformof" },
+            { text: "ISORIGINALFORMOF", value: "isoriginalformof" },
+            { text: "ISIDENTICALTO", value: "isidenticalto" },
+            { text: "ISREVIEWEDBY", value: "isreviewedby" },
+            { text: "REVIEWS", value: "reviews" },
+            { text: "ISDERIVEDFROM", value: "isderivedfrom" },
+            { text: "ISSOURCEOF", value: "issourceof" },
+            { text: "ISREQUIREDBY", value: "isrequiredby" },
+            { text: "REQUIRES", value: "requires" },
+            { text: "ISOBSOLETEDBY", value: "isobsoletedby" },
```
Comment (fenekku) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js:184]: Do we want spaces between the `text` values e.g. `ISOBSOLETEDBY` -> `Is obsoleted by` or are curators more familiar with the all caps together?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js [hunk 2]
```diff
@@ -157,13 +228,14 @@ export class RDMDepositForm extends Component {
...
                 />
+                <PublicationDateField required />
+
                 <ContributorsField
                   options={this.vocabularies.metadata.contributors}
                 />
                 <DescriptionsField
                   options={this.vocabularies.metadata.descriptions}
                 />
-                <PublicationDateField required />
```
Comment (fenekku) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js:166]: The mockup here: https://github.com/inveniosoftware/product-rdm/issues/3 puts `PublicationDateField` below Contributors and Description.

**Review by fenekku — APPROVED** [review]
Good after points addressed!

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js (hunk 1 above)
Comment (lnielsen) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js:184]: Question: ``text`` is what is displayed to the user? if so, then yes it should be changed.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js (hunk 2 above)
Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js]: Fixed. Set according to mockup.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js (hunk 1 above)
Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposits/RDMDepositForm/RDMDepositForm.js]: Changed to human-readable format 😇

**Comment by ppanero** [comment]
Blocked merged due to dependent Reacto modules

**Pull request merged**